clientserver/src/mill/clientserver/Server.scala

```diff
@@ -37,18 +37,32 @@ class Server[T](lockBase: String,
       var running = true
       while (running) {
         Server.lockBlock(locks.serverLock){
-          new File(lockBase + "/io").delete()
-          val ioSocket = new UnixDomainServerSocket(lockBase + "/io")
-          val sockOpt = Server.interruptWith(
-            acceptTimeout,
-            new UnixDomainSocket(lockBase + "/io").close(),
-            ioSocket.accept()
-          )
+          val (serverSocket, sockOpt) = if (ClientServer.isWindows) {
+            val socketName = """\\.\pipe\""" + new File(lockBase).getName
+            val ioSocket = new Win32NamedPipeServerSocket(socketName)
+            (ioSocket, Server.interruptWith(
```

Comment:
Looks like most of the `Server.interruptWith` block is the same, except for the difference in the `new FooSocket` logic; could you limit the if-statement to initializing the ioSocket & a `close` callback, and leave the `sockOpt` definition outside the if-statement?

Reply:
Sure.